08-rails/debugging-rails-vscode.md

```diff
@@ -0,0 +1,118 @@
+
+  $: bundle show ruby-debug-ide
+  #returns /path/to/rubygem/gems/ruby-debug-ide-x.x.x
+
+  $: which rspec
+  #returns /path/to/rubygem/bin/rspec
+```
+
+We will need the output from the commands from above in just a moment in order to tell VS Code how to talk to these gems.
+
+Open the debugging panel using ⇧⌘D, and click on the drop down menu near the play icon. Select `Add New Configuration`.
+
+![alt-text](images/rails-debugger-config-setup.png "The cursor is highlighting the Add New Configuration option on a drop down menu in VS Code's debugging menu")
+
+The next selections you make are somewhat arbitrary, as we will be replacing the code that this action generates with our own custom version.
+
+Once you have the `launch.json` file loaded and open, copy the code below and paste those three paths into the appropriate locations. You should paste each path in two different spots in the file (please remove the asterisks).
+
+```json
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [
+        
+        {
+            "name": "Start Rails server",
+            "type": "Ruby",
+            "request": "launch",
+            "cwd": "${workspaceRoot}",
+            "program": "${workspaceRoot}/bin/rails",
+            "args": [
+                "server",
+                "-p",
+                "3000"
+            ]
+        },
+        {
+            "name": "Debug Rails server",
+            "type": "Ruby",
+            "request": "launch",
+            "cwd": "${workspaceRoot}",
+            "useBundler": true,
+            "pathToBundler": "****(which bundle)YOUR/BUNDLE/PATH/HERE****",
+            "pathToRDebugIDE": "****(bundle show ruby-debug-ide)YOUR/RUBY/DEBUG/IDE/HERE****",
+            "program": "${workspaceRoot}/bin/rails",
+            "args": [
+                "server",
+                "-p",
+                "3000"
+            ]
+        },
+        {
+            "name": "Run RSpec - all",
+            "type": "Ruby",
+            "request": "launch",
+            "cwd": "${workspaceRoot}",
+            "program": "****(which rspec)YOUR/RSPEC/PATH/HERE****",
+            "args": [
+                "--pattern",
+                "${workspaceRoot}/spec/**/*_rspec.rb"
+            ]
+        },
+        {
+            "name": "Debug RSpec - open spec file",
+            "type": "Ruby",
+            "request": "launch",
+            "cwd": "${workspaceRoot}",
+            "useBundler": true,
+            "pathToBundler": "****YOUR/BUNDLE/PATH/HERE****",
+            "pathToRDebugIDE": "****YOUR/RUBY/DEBUG/IDE/HERE****",
+            "debuggerPort": "1235",
+            "program": "****YOUR/RSPEC/PATH/HERE****",
+            "args": [
+                "${file}"
+            ]
+        }
+    ]
+}
```

Comment:
I kind of want to write a script to generate this file since the copy/pasting is kind of tedious.  (Though then we have to get the students to install a script so *shrug*.)

Reply:
Do we want to tell people how to add the script to their path or just keep it simple and have them copy around a `.rb` file?